### PR TITLE
[Manual] [Backport 2.x] [Look&Feel] Replace browser tooltip usage with OUI tooltip

### PR DESCRIPTION
**Pull request by danieldong51**
### Description

Manual backport #7231 to 2.x. Conflicts include:
- `src/plugins/advanced_settings/public/header_user_theme_menu.tsx` due to the file not existing in 2.x (See #6978)
- `src/core/public/chrome/ui/header/home_loader.tsx` regarding `import Url from 'url'` (See #2910). 

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by zhongnansu — COMMENTED** [review]
can you fix the sanpshot failure in CI?
image

File: src/plugins/data/public/ui/filter_bar/filter_options.tsx [hunk 1]
```diff
@@ -28,7 +28,13 @@
  * under the License.
  */
 
-import { EuiButtonIcon, EuiContextMenu, EuiPopover, EuiPopoverTitle } from '@elastic/eui';
+import {
+  EuiButtonIcon,
```
Comment (virajsanghvi) [inline src/plugins/data/public/ui/filter_bar/filter_options.tsx:32]: This was EuiSmallButtonIcon in the previous diff - why change it in the backport?

File: src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx [hunk 2]
```diff
@@ -169,21 +170,21 @@ export function SavedQueryManagementComponent({
...
-      title={i18n.translate('data.search.searchBar.savedQueryPopoverButtonText', {
-        defaultMessage: 'See saved queries',
-      })}
-      data-test-subj="saved-query-management-popover-button"
-    >
-      <EuiIcon type="save" className="euiQuickSelectPopover__buttonText" />
-      <EuiIcon type="arrowDown" />
-    </EuiButtonEmpty>
+    <EuiToolTip content={label} delay="long" position="bottom">
+      <EuiButtonEmpty
```
Comment (virajsanghvi) [inline src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx:179]: Similar, this was EuiSmallButtonEmpty

File: src/plugins/discover/public/application/components/sidebar/discover_field_bucket.tsx [hunk 3]
```diff
@@ -29,7 +29,14 @@
  */
 
 import React from 'react';
-import { EuiText, EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
+import {
+  EuiText,
+  EuiButtonIcon,
```
Comment (virajsanghvi) [inline src/plugins/discover/public/application/components/sidebar/discover_field_bucket.tsx:34]: Was EuiSmallButtonIcon

**Review by virajsanghvi — COMMENTED** [review]
Did [header_user_theme_menu.tsx](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7231/files#diff-d33948eb070fd8889f9482beac714f540ef766fba73cd2d7fa43ace6e342f00b) go away?

**Review by virajsanghvi — DISMISSED** [review]
This looks good given differences are because other changes aren't in 2.x, but we should validate the missing file doesn't intend to get to 2.x and that the miki's changes will still get picked up when backported. May make sense to hold off for those changes if they're coming soon.

**Review by ananzh — APPROVED** [review]

**Comment by zhongnansu** [comment]
@danieldong51 could you check and resolve the conflicting files?

**Comment by danieldong51** [comment]
ciGroup3 failing does not look related to my changes, rather due to EuiTextArea being changed to EuiCompressedTextArea in `src/plugins/data_source_management/public/components/direct_query_data_sources_components/direct_query_data_source_configuration/prometheus/configure_prometheus_data_source.tsx`

**Review by LDrago27 — APPROVED** [review]

**Review by ananzh — APPROVED** [review]

**Pull request merged**